### DISCUSSION TITLE
CBETA API與CBReader轉檔格式問題

### DISCUSSION THREAD

**Discussion by MrMYHuang** [DILA-edu/cbeta-api#18] [Q&A]
各位大德您好：

我最近為了讓我開發的cbetar2 app能支援CBETA離線資料庫，自行仿照CBETA API刻一個基本版的後端程式(cbetar2 backend),目前主要支援目標是：目錄瀏覧及經文顯示(XML to HTML)。
XML to HTML部分我是仿照CBReader cbxml.cpp
https://github.com/heavenchou/CBReader/blob/master/src/cbxml.cpp
使用XSLT重寫一個XSLT程式 - tei.xsl（因為我覺得要把cbxml.cpp抽出、整合至我的app可能有困難，主要是跨平台的第3方libraries編譯問題）

由於我之前寫的cbetar2前端app (cbetar2 frontend)，接的是CBETA API。我原以為CBETA API也是用cbxml.cpp作XML to HTML，這樣我用tei.xsl作XML to HTML的輸出應該會相容，我的cbetar2 frontend就可以順利的在CBETA API與cbetar2 backend之間作切換。

但目前發現CBETA API的HTML與cbxml.cpp的HTML有不同處，例如搜尋CBReader原始碼，找不到輸出`<span class="lb" ...`的程式碼...

請問是否可能把CBETA API的XML to HTML的程式碼模組化並開放此模組原始碼，以方便第3方使用或改寫？

**Comment by RayCHOU** [comment]
> 請問是否可能把CBETA API的XML to HTML的程式碼模組化並開放此模組原始碼，以方便第3方使用或改寫？

我想過這問題，除了覺得程式有點亂，不除斥公開。
請問 @jenjouhung 的意見？

註： CBETA API 的程式是用 Ruby。

**Comment by ghost** [comment]
可以公開啊，不過我也可以想像那有多複雜

'Ray Chou' via da_對外 <da@dila.edu.tw> 於 2021年2月4日 週四 10:37 寫道：

**Reply by MrMYHuang** [reply]
謝謝各位🙏

**Comment by RayCHOU** [comment]
CBETA API 網站的 Source 已公開，
XML 轉 HTML for UI 的程式請參考: https://github.com/DILA-edu/cbeta-api/blob/main/lib/tasks/x2h_for_ui.rb

**Reply by MrMYHuang** [reply]
謝謝您
